### PR TITLE
Ensure files end with a newline with Spotless rather than Checkstyle

### PR DESCRIPTION
**Pull request by basil**
Using [Checkstyle](https://checkstyle.sourceforge.io/) to ensure that there is a newline at the end of each file is not ideal because Checkstyle offers no way to automatically fix violations. In contrast, [Spotless](https://github.com/diffplug/spotless) offers both a `check` goal to report violations as well as an `apply` goal to automatically fix all violations, which is far more convenient.

In this PR we disable the [`NewlineAtEndOfFile`](https://checkstyle.sourceforge.io/config_misc.html#NewlineAtEndOfFile) Checkstyle check and replace it with the `endWithNewline` Spotless check. The Spotless `check` goal runs in the `verify` phase by default and fails if there are any violations. The developer can then run `mvn spotless:apply`, which automatically fixes all violations.

Note that while the Checkstyle version of this check enforced that there were one or more newlines at the end of each file, the Spotless version enforces that there is exactly one newline at the end of each file. So this contains a few changes to remove duplicate newlines as well.

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
